service/auction.go

```diff
@@ -0,0 +1,141 @@
+	}
+	// order.Size pending should be greater or equal to orderFrag: (Order.sizePending + prder.pending) >= frag.size
+	return order.SizePending.GreaterThanOrEqual(frag.Size)
+}
+
+func (s *Service) ConfirmAuction(ctx context.Context, auctionId uuid.UUID) (ConfirmAuctionRes, error) {
+	// TODO: re-entrance validate it doesnt already confirmed
+
+	// get auction from store
+	frags, err := s.orderBookStore.GetAuction(ctx, auctionId)
+	if err != nil {
+		logctx.Warn(ctx, err.Error())
+		return ConfirmAuctionRes{}, err
+	}
+
+	res := ConfirmAuctionRes{}
+
+	// validate all orders of auction
+	for _, frag := range frags {
+		// get order by ID
+		order, err := s.orderBookStore.FindOrderById(ctx, frag.OrderId, false)
+		if order == nil {
+			// cancel auction
+			s.orderBookStore.RemoveAuction(ctx, auctionId)
+
+			// return empty
+			logctx.Warn(ctx, err.Error())
+			return ConfirmAuctionRes{}, models.ErrOrderNotFound
+		} else if !validateOrderFrag(frag, order) {
+			// cancel auction
+			s.orderBookStore.RemoveAuction(ctx, auctionId)
+
+			// return empty
+			logctx.Warn(ctx, err.Error())
+			return ConfirmAuctionRes{}, models.ErrAuctionInvalid
+		} else {
+			// success- append
+			res.Orders = append(res.Orders, order)
+			res.Fragments = append(res.Fragments, &frag)
+		}
+	}
+	// set order order fragments as Pending
+	for i := 0; i < len(res.Orders); i++ {
+		// lock frag.Amount as pending per order - no STATUS_PENDING is needed
+		res.Orders[i].SizePending = res.Fragments[i].Size
+	}
+	s.orderBookStore.StoreOrders(ctx, res.Orders)
+
+	// add oredebook signature on the buffer
+	res.BookSignature = []byte("todo:sign")
+
+	return res, nil
+}
+
+func (s *Service) RevertAuction(ctx context.Context, auctionId uuid.UUID) error {
+	// get auction
+
+	// for each order-frag, revert the pending in original order
+
+	return nil
+}
+
+func (s *Service) AuctionMined(ctx context.Context, auctionId uuid.UUID) error {
+	// TODO: re-entrance validate it doesnt already confirmed
+
+	// get auction from store
+	frags, err := s.orderBookStore.GetAuction(ctx, auctionId)
+	if err != nil {
+		logctx.Warn(ctx, err.Error())
+		return err
+	}
+	var filledOrders []*models.Order
+
+	// validate all pending orders fragments of auction
+	for _, frag := range frags {
+		// get order by ID
+		order, err := s.orderBookStore.FindOrderById(ctx, frag.OrderId, false)
+		if order == nil {
+			// cancel auction
+			s.orderBookStore.RemoveAuction(ctx, auctionId) // PANIC - shouldn't happen
+
+			// return empty
+			logctx.Error(ctx, err.Error())
+			return models.ErrOrderNotFound
+		} else if !validatePendingFrag(frag, order) {
+			// cancel auction
+			s.orderBookStore.RemoveAuction(ctx, auctionId) // PANIC - shouldn't happen
+
+			logctx.Error(ctx, fmt.Sprintf("validatePendingFrag failed. PendingSize: %s FragSize:%s", order.SizePending.String(), frag.Size.String()))
+			return models.ErrAuctionInvalid
+		} else {
+			// fill fragment in the order
+			order.SizePending.Sub(frag.Size)
```

Comment:
Is this right? Shouldn't we add to ```SizePending```? And we know when an order has been fully filled when ```SizeFilled == Size```?